pymeasure/instruments/teledyne/teledyneT3AFG.py

```diff
@@ -0,0 +1,118 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_range, strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class SignalChannel(Channel):
+    
+    output_enabled = Channel.control(
+        "C{ch}:OUTPut?",
+        "C{ch}:OUTPut %s",
+        """Control whether the channel output is enabled (boolean).""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 'ON', False: 'OFF'},
+        get_process=lambda x: x[0].split(' ')[1],
+        # Not sure why "OFF" doesn't get transformed into False
```

Comment:
Consider writing a test for this, which will allow you to compare expectation to your code.
Verify, that your device responds with "OFF" (in that part of the message).

Reply:
The string returned by the instrument is quite long, but I put an example in the test for this control.

I'm not sure that I'm doing the mapping right though. The result of the get_process string parsing lambda does not appear to go through the "map_values" mapping before being returned to the user. I just get "OFF" as opposed to my expected False. Are get_process and map_values mutually exclusive?

Reply:
Oh yes, right now `map_values` works only, if `values` returns a single value. #740 tries to fix this.

You can do it in the get_process, if you like:
```python
        get_process=lambda x: :True if [0].split(" ")[1] == "ON" else False
```

Reply:
I went with that suggestion, thank you!